### ISSUE TITLE
Libre Barcode 128 font is missing code character number 92

### ISSUE THREAD

**Issue by DavinChurch** [google/fonts#6877]
The "Libre Barcode 128" font is missing code character number 92 (ASCII 124 = x007C = 'fs' = '¦'). I do not need to encode text with this character but it is critical as a check character. Instead of a barcode, it simply prints "¦".

Try encoding the text "PRO222631" in Code128b and you will get the missing character as the check character.

It should print an appropriate bar code (widths = 111143) instead of the ASCII symbol.

Example:
![Barcode128 error](https://github.com/google/fonts/assets/55806407/b9188321-a038-4b21-aba0-0d909b873727)

**Comment by graphicore** [comment]
Hi @DavinChurch, thanks for reporting.

I don't know how you encoded your input for Libre Barcode 128, but the symbol 92 in this font is encoded at `|` (VERTICAL LINE), which is also ASCII/Unicode `124` or `U+007C` Wikipedia [Vertical Bar](https://en.wikipedia.org/wiki/Vertical_bar).

The font is implemented after the table in https://en.wikipedia.org/wiki/Code_128#Bar_code_widths

`¦` (BROKEN BAR ) is in Unicode `166` or `U+00A6` 

But that Wikipedia article about the Vertical Bar also has a paragraph about [Solid vertical bar vs broken bar](https://en.wikipedia.org/wiki/Vertical_bar#Solid_vertical_bar_vs_broken_bar) maybe some of the confusion happens in Windows Notepad, is it set to UTF-8 encoding? I don't have access.

**Comment by DavinChurch** [comment]
Hello @graphicore  - thanks for your reply! What I was sending to the font was an ASCII 124 but the broken vertical bar is what appeared as output. This was true in several output mediums, all using the font. I suppose it is possible that Notepad converted it to the broken bar in some way for my example, but it seems unlikely that all my different methods would have the same sort of problem with the same font, including my Dymo label printer which renders it as the usual "open box" character instead of the proper barcode.

It's also possible that I'm doing something wrong, or that the OS or driver is intercepting my character before the font gets it, but everything else seems to be working fine. I don't suppose I should be using some sort of alternate character to print that symbol, should I?

I can't think of what else might be happening, can you?

**Comment by DavinChurch** [comment]
Oh, @graphicore, one other thing I thought of is that perhaps Windows is simply substituting another font to show me that character if it doesn't find (or thinks it doesn't find) the one in the requested font. That probably explains what the character is doing there, if it isn't actually in your font (which I suspect is true).

**Comment by graphicore** [comment]
@DavinChurch 

> I can't think of what else might be happening, can you?

In a web-browser, I get the correct symbol using `|`, but that's a proper utf-8 environment. works even in https://fonts.google.com/specimen/Libre+Barcode+128+Text?preview.text=%7C

> I can't think of what else might be happening, can you?

How do you encode the data, i.e. calculate the charcode etc. of the checksum symbol? Can you post the encoded string you get as a result? Does it work in [google fonts](https://fonts.google.com/specimen/Libre+Barcode+128+Text)?

> perhaps Windows is simply substituting another font to show me that character if it doesn't find

Yes, very much possible, but it should still rather be the vertical bar and not the broken bar. Some software (also web browsers) fall back to another font to find not defined chars, others use the `.notdef` glyph, which usually is the "open box".

There's also an encoder at https://graphicore.github.io/librebarcode/documentation/code128.html but it doesn't produce just code128b for you, however I brute forced input by hand to get a `|` checksum symbol. Value `PROvfrxzxsdfw2` encoded `ÌPROvfrxzxsdfw2|Î` in [google fonts](https://fonts.google.com/specimen/Libre+Barcode+128+Text?preview.text=%C3%8CPROvfrxzxsdfw2%7C%C3%8E)

![Screenshot from 2023-10-20 16-46-52](https://github.com/google/fonts/assets/393132/a4aaf070-a21f-4c76-b028-8d258a276003)

**Comment by DavinChurch** [comment]
Well, I'm mystified. I'm using the equivalent of CHAR(124) to output it, so it seems difficult to get anything else. I've checked the checksum formula and it is producing a 92 checksum, as confirmed by several on-line utilities.

Aha!! I switched from a non-Unicode application to a pure-Unicode one and found that it was now generating an UNBROKEN bar. This now represents correctly in your font, so apparently the problem is in the non-Unicode application converting a binary value internally when it is supposed to be leaving it strictly alone. I'll put together a workaround for that.

In the meantime, may I suggest that you duplicate the glyph in the font so that either vertical bar shows up correctly? This would be conceptually similar to providing an alternate space-character at 194 instead of 32 for systems that don't handle the 32 as a glyph.

Other than that, I'm happy to call this issue resolved and I'll deal with the application on my end. Thank you very much!!
